naga/src/proc/mod.rs

```diff
@@ -297,6 +297,9 @@ impl super::TypeInner {
             } => {
                 let count = match size {
                     super::ArraySize::Constant(count) => count.get(),
+                    // any struct member or array element needing a size at pipeline-creation time
+                    // must have a creation-fixed footprint
+                    super::ArraySize::Pending(_) => 0,
```

Comment:
Ideally this would be `unreachable!();` to catch bugs early.

Reply:
The dynamic case is reachable and valid, for the purposes of layout calculations dynamic arrays act as an array with a single element.

Reply:
Hm, I do see that this might be hard to make `unreachable!();`, if it's too involved we can keep it as is.

Reply:
Oh, you're right about the dynamic arrays, Maybe the `BindingArray` option below that returning 0 is the better example? I think making it unreachable would mostly consist of pushing

https://github.com/gfx-rs/wgpu/blob/trunk/naga/src/proc/layouter.rs#L170

into the match arms and not calling it for Array where the size is pending.

Reply:
Ah I see there is precedent for returning 0. Let's keep it as is.